src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int index = findIndexByKey(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        } else if (key == node.key || (key != null && key.equals(node.key))) {
+            table[index].value = value;
+            return;
+        }
+        while (node.next != null) {
+            if (key == node.next.key || (key != null && key.equals(node.next.key))) {
+                node.next.value = value;
+                break;
+            }
+            node = node.next;
+        }
+        if (node.next == null) {
+            node.next = new Node<>(key, value, null);
+            size++;
+        }
```

Comment:
seems to be very complicated, lets try to do smth like that
```suggestion
        } else {
            while (node != null) {
                if (Objects.equals(node.key, key)) {
                    node.value = value;
                    return;
                }
                if (node.next == null) {
                    node.next = newNode;
                    break;
                }
                node = node.next;
            }
        }
```